stdlib/domain.ml

```diff
@@ -25,6 +25,8 @@ module Raw = struct
     = "caml_ml_domain_id"
   external cpu_relax : unit -> unit
     = "caml_ml_domain_cpu_relax"
+  external get_recommended_domains: unit -> int
```

Comment:
This function may be marked as [@@noalloc]
```suggestion
  external get_recommended_domains: unit -> int [@@noalloc]
```

Reply:
Not specific to this PR, but `cpu_relax` and `self` should also be made [@@noalloc].